services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/client/pulsar/DataPlaneClientImpl.java

```diff
@@ -18,24 +18,151 @@
 import com.futurewei.alcor.dataplane.client.DataPlaneClient;
 import com.futurewei.alcor.dataplane.entity.MulticastGoalState;
 import com.futurewei.alcor.dataplane.entity.UnicastGoalState;
+import com.futurewei.alcor.dataplane.exception.GroupTopicNotFound;
+import com.futurewei.alcor.dataplane.exception.MulticastTopicNotFound;
 import com.futurewei.alcor.schema.Goalstate;
 import com.futurewei.alcor.schema.Goalstateprovisioner;
+import com.futurewei.alcor.web.entity.dataplane.MulticastGoalStateByte;
+import com.futurewei.alcor.web.entity.dataplane.UnicastGoalStateByte;
+
+import org.apache.pulsar.client.api.Producer;
+import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.impl.schema.JSONSchema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
 
+import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
 //@Component
 public class DataPlaneClientImpl implements DataPlaneClient {
+    private static final Logger LOG = LoggerFactory.getLogger(DataPlaneClientImpl.class);
+
+    @Autowired
+    private PulsarClient pulsarClient;
+
+    @Autowired
+    private TopicManager topicManager;
 
     @Override
     public Map<String, List<Goalstateprovisioner.GoalStateOperationReply.GoalStateOperationStatus>> createGoalStates(Goalstate.GoalState goalState, String hostIp) throws Exception {
         return null;
     }
 
+    private UnicastGoalStateByte buildUnicastGoalStateByte(UnicastGoalState unicastGoalState) {
+        UnicastGoalStateByte unicastGoalStateByte = new UnicastGoalStateByte();
+        unicastGoalStateByte.setNextTopic(unicastGoalState.getNextTopic());
+        unicastGoalStateByte.setGoalStateByte(unicastGoalState.getGoalState().toByteArray());
+
+        return unicastGoalStateByte;
+    }
```

Comment:
This basically converts UnicastGoalState to UnicastGoalStateByte. Can we define it inside the class of UnicastGoalState?